2002 uta deanna huynh form validation

sfjsss opened the pull request:
-refactor front end form validation using angular built-in form validation
-render success/error message based on JSON response we received upon the form submission
-refractor form so users can submit forms with ENTER key
-fixed the updateIsDriver request in carService by removing the userId so it doesn't send a URL that our back-end controller doesn't know and violate CORS
-remove {observe: "response"} from the addCar request in carService so we can receive the proper JSON response to check the validation information
-remove the logic which is trying to duplicate "home address" into "work address" in sign up modal component to prevent back-end controller receiving duplicate data

deannahuynh committed:
added form validations login, sign-up, profile-car

deannahuynh committed:
added form validations login, sign-up, profile-car

deannahuynh committed:
added form validations on all profile components

deannahuynh committed:
added form validations on all profile components

deannahuynh committed:
removed wAddress field in sign up modal

deannahuynh committed:
removed wAddress field in sign up modal

deannahuynh committed:
disabled sumbit with empty fields on sign up

deannahuynh committed:
disabled sumbit with empty fields on sign up

deannahuynh committed:
disabled sumbit with empty fields on sign up

deannahuynh committed:
disabled sumbit with empty fields on sign up

sfjsss committed:
add missing } in sign-up-modal.component.ts

sfjsss committed:
add missing } in sign-up-modal.component.ts

sfjsss committed:
resolve conflict in profile-location.component.html

sfjsss committed:
resolve conflict in profile-location.component.html

sfjsss committed:
integrate back-end validation

sfjsss committed:
integrate back-end validation

sfjsss committed:
integrate back-end validation in profile-contact component

sfjsss committed:
integrate back-end validation in profile-contact component

sfjsss committed:
integrate back-end validation in profile-location component

sfjsss committed:
integrate back-end validation in profile-location component

chrisrchar committed:
updated car and user models to reflect change in backend model's relationship. Also fixed files that referenced the models

sfjsss committed:
integrate back-end validation in profile-car component

sfjsss committed:
integrate back-end validation in profile-car component

chrisrchar committed:
Merge pull request #8 from Batch-747-Carolyn-Rehm/2002-uta-ChristianRandle-UserRefactorFrontend

updated car and user models to reflect change in backend model's rela…

sfjsss committed:
resolve conflicts

sfjsss committed:
default car field to be null when create new user to avoid referencing error

carolynrehm commented on src/app/components/sign-up-modal/sign-up-modal.component.ts:
```diff
@@ -104,84 +109,47 @@ openModal(template :TemplateRef<any>){
+  this.userService.addUser(this.user).subscribe(
+    res => {
+      if (Object.keys(res).length === 0) {
+        this.http.get(`${environment.loginUri}?userName=${this.user.userName}&passWord='placeholder'`).subscribe(response => {
```
This HTTP call should be moved to a service

sfjsss replied:
removed this request to authService

carolynrehm reviewed (commented):
Filtering and sorting in src/app/components/driver-info/driver-info.component.ts is fine for now utilizing the legacy code but that functionality should be moved to the backend eventually

sfjsss committed:
update createCarInfo in profile-car component so now it will update user to create a car

sfjsss committed:
move login authenticate http request from sign up modal component to authService

sfjsss commented:
-change createCarInfo function in profile-car component to send updateUserInfo to fix the issue where car object was unable to establish a relationship with the user
-remove authenticate http request from sign up modal component to authService

sfjsss committed:
fix bugs on profile membership component

sfjsss commented:
1. the success message p tag was inside of the seats available select tag hence it doesn't render properly, i moved it out
2. the updateCarInfo and updateUserInfo calls inside of updateMembershipInfo function were not subscribed, and because httpClient by default returns an observable, if we don't subscribe it, the request will never get send, hence the membership form didn't work at all.
3. calling two services here doesn't really make sense since we can just update the user object, and attach the updated car object to it, and just update userInfo. Having two services call also messed up the updating logic because user object by default has null on car field, and it will override carInfoUpdate. I combined the two service calls by attaching updated car object to user object then just call updateUserInfo.
4. the updateCarInfo function inside of carService is sending the put method to localhost:8080/cars/carId. The carId is not used at all since we use .save(user) and user object already contains the car object with carId. carId also caused potentially issue because new user will not have a car which makes the carObject null, which creates null pointer exception. So I changed the url to localhost:8080/cars instead

jaybouche reviewed (approved):
code added works. CircleCI issues addressed in other pending branch merge.

jbennatt reviewed (approved):
Car save works fully. Membership page appears to work for newly created users but not users created from data.sql.
